logstash-core/src/test/java/org/logstash/common/clients/HttpClientTest.java

```diff
@@ -0,0 +1,371 @@
+package org.logstash.common.clients;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.junit.WireMockRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.logstash.common.clients.HttpClient.RequestFailedException;
+
+import java.nio.file.Paths;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.options;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+
+public class HttpClientTest {
+
+    @Rule
+    public WireMockRule httpServer = new WireMockRule(options().dynamicPort());
+
+    @Test
+    public void canMakeHttpRequestWithAlmostDefaultSettings() throws Exception {
+        final String path = "/api/hello";
+        final String expectedResponseBody = "Hello, World";
+
+        httpServer.stubFor(get(urlPathEqualTo(path))
+                .willReturn(aResponse()
+                        .withStatus(200)
+                        .withBody(expectedResponseBody))
+        );
+
+        HttpClient httpClient = HttpClient.builder()
+                .port(httpServer.port()) // We set this one setting so we don't need to run this test as a superuser
+                .build();
+
+        assertThat(httpClient.get(path)).isEqualTo(expectedResponseBody);
+    }
+
+    @Test
+    public void canMakeHttpRequestWithCustomHostnameAndPort() throws Exception {
+        final String BIND_ADDRESS = "127.0.0.1";
+
+        WireMockServer httpServer = new WireMockServer(options()
```

Comment:
NIT: I'd probably give this variable a different name, since you have a field `httpServer ` as well, that took me 2 seconds to read :)